aggregators/aggregator.go

```diff
@@ -523,6 +546,29 @@ func (a *Aggregator) harvestForInterval(
 			len(errs), cmCount, errors.Join(errs...),
 		))
 	}
+
+	// All remaining events in the cached events map should be failed events.
+	// Record these events with a failuer outcome.
+	for cmID, eventsTotal := range cachedEventsStats {
+		if eventsTotal < 0 {
+			a.cfg.Logger.Warn(
+				"unexpectedly failed to harvest all collected events",
+				zap.Duration("aggregation_interval_ns", ivl),
+				zap.ByteString("id", cmID[:]),
+				zap.Float64("remaining_events", eventsTotal),
+			)
+			continue
```

Comment:
[**For reviewers**] This should not happen. I have logged it as an `warning` here so that if there is a bug that causes such a case then our SLOs are not impacted and we have a warning log. Let me know if there is a different suggestion to handle this case.